client/my-sites/checkout/checkout/test/country-specific-payment-fields.js

```diff
@@ -21,23 +24,38 @@ const defaultProps = {
 };
 
 describe( '<CountrySpecificPaymentFields />', () => {
-	test( 'should render', () => {
-		const wrapper = shallow( <CountrySpecificPaymentFields { ...defaultProps } /> );
-		expect( wrapper ).toMatchSnapshot();
-	} );
+	test( 'should call this.props.handleFieldChange when updating field', async () => {
+		const user = userEvent.setup();
+
+		const WrappedCountrySpecificPaymentFields = ( props ) => {
+			const [ fields, setFields ] = useState( {} );
+
+			props.getFieldValue.mockImplementation( ( name ) => {
+				return fields[ name ];
+			} );
+
+			props.handleFieldChange.mockImplementation( ( name, value ) => {
+				setFields( { ...fields, [ name ]: value } );
+			} );
+
+			return <CountrySpecificPaymentFields { ...props } />;
+		};
 
-	test( 'should call this.props.handleFieldChange when updating field', () => {
-		const wrapper = shallow( <CountrySpecificPaymentFields { ...defaultProps } /> );
-		const documentInput = wrapper.find( '[name="document"]' );
-		const event = { target: { name: 'document', value: 'spam' } };
-		documentInput.simulate( 'change', event );
-		expect( defaultProps.handleFieldChange ).toBeCalledWith( 'document', 'spam' );
+		render( <WrappedCountrySpecificPaymentFields { ...defaultProps } /> );
+
+		const box = screen.getByRole( 'textbox', { name: /taxpayer identification number/i } );
+		await user.type( box, 'spam' );
+
+		expect( defaultProps.handleFieldChange ).toHaveBeenNthCalledWith( 4, 'document', 'spam' );
```

Comment:
We could as well make this:

```suggestion
		expect( defaultProps.handleFieldChange ).toHaveBeenCalledWith( 'document', 'spam' );
```

to make it less specific if we don't care so much if it triggered 4 times or not (whether change triggering is debounced or not)

Reply:
Agreed 👍🏻